muffnn/mlp/base.py

```diff
@@ -252,11 +261,22 @@ def _set_up_graph(self):
                 if self.keep_prob != 1.0:
                     t = tf.nn.dropout(t, keep_prob=self._keep_prob)
                 t = affine(t, layer_sz, scope='layer_%d' % i)
+
             t = t if self.activation is None else self.activation(t)
 
+            # Set transformed layer to hidden layer
+            if self._transform_layer_index == i:
+                self._transform_layer = t
+
         # The output layer and objective function depend on the model
         # (e.g., classification vs regression).
         t = self._init_model_output(t)
+
+        # set the transform layer to the output logits if it is not set
+        # already.
+        if not hasattr(self, '_transform_layer'):
```

Comment:
It might be nice to raise a `ValueError` here or at the start of this function if the user mistakenly provides a bad value for `transform_layer_index`.  Otherwise, they might unexpectedly get the output layer when they wanted something else.

Reply:
Will do.